src/types/captcha.ts

```diff
@@ -64,12 +65,8 @@ export type CaptchaSolution = {
 }
 
 export type CaptchaConfig = {
-    solved: {
-        count: number
-    },
-    solvedAndUnsolved: {
-        count: number
-    }
+    state: CaptchaStates,
```

Comment:
Sorry @muzammil-gimly I think I went off on a tangent in that review. Re-reading makes me think another change is required. This is due to unclear requirements so apologies.

You were nearly there with your initial config. The only thing that needs to change is switching `solvedAndUnsolved` to `unsolved`

```
   solved: {
        count: number
    },
    unsolved: {
        count: number
    }
```
Then later on you simply need the following two lines to get the various captchas:

```
const solved = await this.getCaptchaWithProof(datasetId, true, this.captchaConfig.solved.count)
const unsolved = await this.getCaptchaWithProof(datasetId, true, this.captchaConfig.unsolved.count)
```

This allows the user to decide how many solved and how many unsolved they want per request. It removes the ened for balancing 50:50 with `n/2` etc.

Reply:
Bear in mind that the count must be a whole number.

Reply:
And the use should not be allowed to put `n=0` `solved` and `n>0` `unsolved` as this makes no sense.